### DISCUSSION TITLE
Adafruit DPS310 barometer not recognized

### DISCUSSION THREAD

**Discussion by clarkpark** [iNavFlight/inav#8193] [Q&A] [answered]
I cannot get my Foxeer F722 V2 FC (no internal baro) to see my external Adafruit DPS310 barometer using INAV 5.0.0.  I have tried both 0x77 and 0x76 addresses.  I am passing magnetometer I2C signal through the Adafruit board, and the FC is able to see the magnetometer OK, but not the barometer.  I have the barometer set to DPS310 on the configuration tab.

**Comment by JulioCesarMatias** [comment]
Check that the SDA and SCL connections are correct... Provide more information, such as photos, electrical diagram, etc...

**Reply by clarkpark** [reply]
Yes, SDA and SCL are correct.  Else I wouldn't be able to see the magnetometer passing through the board.  No need for wiring diagram, because SDA/SCL from compass goes to Adafruit board, SDA/SCL/5V/Ground from Adafruit board to corresponding pins on FC.  Compass works, baro does not.

What address does INAV use for the DPS310 baro?

**Comment by JulioCesarMatias** [comment]
0x76

**Comment by JulioCesarMatias** [comment]
Try to increase the working speed of the I2C bus from 400Khz to 800Khz, maybe something will change...

**Comment by clarkpark** [comment]
I tried all I2C speed options without luck, with and without compass connected.  Board is set to 0x76 via solder bridge.  Attached is photo of the wiring.  Cable on the left goes to compass.
![PXL_20220704_210259772](https://user-images.githubusercontent.com/108638075/177217100-437ce51c-d92a-46d9-b9de-c791e7bf7e3c.jpg)

**Comment by JulioCesarMatias** [comment, marked as answer]
Problem found! The DPS310 barometer drivers are not active on this target. Only the BMP280 or MS5611. You can edit the algorithm, and compile with the DPS310 driver active!

![Captura de tela 2022-07-04 184006](https://user-images.githubusercontent.com/25625982/177218696-ec0b5bcf-44f8-4b94-8b6b-0fd134d05705.jpg)

**Reply by clarkpark** [reply]
Great!
Can this be fixed or do I need to buy another barometer?

**Reply by JulioCesarMatias** [reply]
You can download the iNav algorithm, and use Visual Studio Code to edit the target file, and add the definition `#define USE_BARO_DPS310` and compile, and load the generated .Hex to your FC.

**Reply by MrD-RC** [reply]
I wonder if the we could limit the lists in Configurator to only show valid hardware. Could an MSP call be used to check defines?

Passing the define to check is easy enough. But can C construct the define to check from a string received via MSP? Or would there need to be a big logic block covering all cases?

**Reply by stronnag** [reply]
If we do this (have an FC API that provides consumers with a list of valid sensors for a target), then it could be also provide a means by which the FC and Configurator are always "in sync" over the lists of supported sensors, and we avoid the situation where a sensor is removed or added in the FC and the configurator still has the old list.

**Reply by clarkpark** [reply]
<img width="367" alt="image" src="https://user-images.githubusercontent.com/108638075/177482625-0f8145a3-b74f-4be8-8f26-4757f02473ae.png">
I downloaded 'inav-master' from github and visual studio code.  Not sure how to proceed to add the DP310.

**Comment by stronnag** [comment]
> I downloaded 'inav-master' from github and visual studio code. Not sure how to proceed to add the DP310.

There is a guide to building custom firmware in the [wiki](https://github.com/iNavFlight/inav/wiki/Building-custom-firmware).

The section on [`target.h`](https://github.com/iNavFlight/inav/wiki/Building-custom-firmware#targeth) is pertinent.  In your case, you would add the define for your barometer to the barometer stanza; `#define USE_BARO_DPS310`. Then compile the firmware for your target. That is all that is necessary.

Then you might submit a Pull Request and join the glorious band of INAV contributors.

**Reply by clarkpark** [reply]
OK, I found the target and edited the file in Visual Studio Code.  I don't understand how to compile.  Sorry I am really out of my element here.
<img width="557" alt="image" src="https://user-images.githubusercontent.com/108638075/177573674-d9dd6bc8-d9c7-4d9d-b71f-14fc55716ee3.png">

**Reply by clarkpark** [reply]
OK, I finally figured out how to compile my modified target.  However, the version is updated to 6.0.0 and the configurator will not work with this release #.  How do I get it to compile to 5.x.x?
